lib/entry/LiveEntry.js

```diff
@@ -75,6 +78,7 @@ function createFlavorDownloaders() {
     var promises = _.map(that.flavorsDownloaders, function(f) {
         f.on('newStreamFiles', onNewTsFilesDownloaded.bind(that));
         f.on('stopped', onFlavorDownloaderStopped.bind(that));
+        f.on('diagnosticsAlert', handleDiagnosticsAlert.bind(that));
```

Comment:
why not calling  
f.on('diagnosticsAlert',  this.diagnostics.handleDiagnosticsAlert.bind(this.diagnostics));